classes/query/QueryDAO.php

```diff
@@ -394,23 +421,71 @@ public function addRecommendationQuery(int $recommenderUserId, int $submissionId
             // Check if the user is unsubscribed
             $notificationSubscriptionSettings = $notificationSubscriptionSettingsDao->getNotificationSubscriptionSettings(
                 NotificationSubscriptionSettingsDAO::BLOCKED_EMAIL_NOTIFICATION_KEY,
-                $discussionParticipantsId,
+                $participantUserId,
                 $contextId
             );
             if (in_array(PKPNotification::NOTIFICATION_TYPE_NEW_QUERY, $notificationSubscriptionSettings)) {
                 continue;
             }
-            $recipient = Repo::user()->get($discussionParticipantsId);
+
+            $recipient = Repo::user()->get($participantUserId);
             $mailable = new Mailable();
             $mailable->to($recipient->getEmail(), $recipient->getFullName());
-            $mailable->from($recommenderUser->getEmail(), $recommenderUser->getFullName());
+            $mailable->from($fromUser->getEmail(), $fromUser->getFullName());
             $mailable->subject($title);
             $mailable->body($content);
+
             Mail::send($mailable);
         }
 
         return $query->getId();
     }
+
+    /**
+     * Create a query with a submission's comments for the editors
+     *
+     * Creates the query and assigns all participants
+     *
+     * @return int new query id
+     */
+    public function addCommentsForEditorsQuery(Submission $submission): int
```

Comment:
Now that we're starting to have queries created by several workflow processes, maybe it's worth adding some kind of a "source" field to differentiate between them in case e.g. we need to later alter the way some of them are stored in the database. We've often had to resort to heuristics to distinguish between things after the fact. (I'm considering a similar addition to the submissions table, for example, to distinguish between imported, quickSubmit, and regular workflow submissions.)

Reply:
Makes sense. I can already think of a use-case here: automatically adding an editor to the comments-for-the-editor discussion when they are assigned after submission. Want me to try to roll that into this PR or file a separate issue?

Reply:
Yes, please file it separately.

Reply:
Filed at https://github.com/pkp/pkp-lib/issues/8367.